src/cpp/rtps/builtin/discovery/participant/PDP.cpp

```diff
@@ -520,9 +520,9 @@ bool PDP::has_writer_proxy_data(const GUID_t& writer)
     {
         if (pit->m_guid.guidPrefix == writer.guidPrefix)
         {
-            for (WriterProxyData* wit : pit->m_writers)
+            for (auto wp : *pit->m_writers)
             {
-                if (wit->guid() == writer)
+                if (wp.second->guid() == writer)
```

Comment:
We could use `pit->m_writers->find` here

Reply:
In fact, we can just do `return pit->m_writers->find(writer.entityId) != pit->m_writers->end();`

Reply:
agreed, we can speed up by using hash table capabilities:
```
 ProxyHashTable<WriterProxyData> & writers = *pit->m_writers;
 return writers.find(writer.entityId) != writers.end();
```